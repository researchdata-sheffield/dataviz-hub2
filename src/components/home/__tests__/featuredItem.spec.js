/**
 * @jest-environment jsdom
 */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from "react";
import { render } from "@testing-library/react";
import FeaturedItem from "../featuredItem";

jest.mock("../../../utils/shared", () => {
  return {
    ...jest.requireActual("../../../utils/shared"),
    getImageSource: () => {
      return null;
    }
  };
});

describe("Home | Featured Items", () => {
  const mockData = {
    edges: [
      {
        node: {
          id: "168c4103-f8fd-58e2-b305-3eb015954844",
          frontmatter: {
            type: null,
            title: "Mock item 1",
            date: "25 January 2021",
            description: "Featured item mock.",
            tag: ["Dataviz", "C++", "Interactive"],
            category: ["Articles"],
            featured: true,
            thumbnail: {
              childImageSharp: {
                gatsbyImageData: {
                  images: {
                    sources: [
                      {
                        srcSet:
                          "/static/8299e3c1cf212758e19d56784b0e4e48/bd3dc/thumb.webp 366w,\n/static/8299e3c1cf212758e19d56784b0e4e48/32213/thumb.webp 732w,\n/static/8299e3c1cf212758e19d56784b0e4e48/8dc67/thumb.webp 1464w",
                        type: "image/webp",
                        sizes: "(min-width: 1464px) 1464px, 100vw"
                      }
                    ]
                  },
                  width: 1464,
                  height: 1125
                }
              }
            },
            author: [
              {
                name: "Mock author 1",
                avatar: {
                  childImageSharp: {
                    gatsbyImageData: {
                      layout: "fullWidth",
                      images: {
                        sources: [
                          {
                            srcSet:
                              "/static/1b950aec9371d81fecb003f21d10c274/5e011/Mock.webp 320w",
                            type: "image/webp",
                            sizes: "100vw"
                          }
                        ]
                      },
                      width: 1,
                      height: 1
                    }
                  }
                }
              }
            ]
          },
          fields: {
            slug: "/blog/25/01/2021/mock-item-1",
            readingTime: {
              text: "11 min read"
            }
          }
        }
      },
      {
        node: {
          id: "006656c2-c336-53f1-936d-a93f8751bba9",
          frontmatter: {
            type: null,
            title: "Mock featured item 2",
            date: "09 September 2020",
            description: "Data visualizations have effects.",
            tag: ["Social Factors", "Effects"],
            category: ["Articles"],
            featured: true,
            thumbnail: {
              childImageSharp: {
                gatsbyImageData: {
                  images: {
                    sources: [
                      {
                        srcSet:
                          "/static/8299e3c1cf212758e19d56784b0e4e48/bd3dc/thumb.webp 366w,\n/static/8299e3c1cf212758e19d56784b0e4e48/32213/thumb.webp 732w,\n/static/8299e3c1cf212758e19d56784b0e4e48/8dc67/thumb.webp 1464w",
                        type: "image/webp",
                        sizes: "(min-width: 1464px) 1464px, 100vw"
                      }
                    ]
                  },
                  width: 1464,
                  height: 1125
                }
              }
            },
            author: [
              {
                name: "Mock author 2",
                avatar: {
                  childImageSharp: {
                    gatsbyImageData: {
                      layout: "fullWidth",
                      images: {
                        sources: [
                          {
                            srcSet:
                              "/static/b6f432ec1b47d139d904d5025714b073/ca9d4/helen_k.webp 84w",
                            type: "image/webp",
                            sizes: "100vw"
                          }
                        ]
                      },
                      width: 1,
                      height: 1.0119047619047619
                    }
                  }
                }
              }
            ]
          },
          fields: {
            slug: "/blog/09/09/2020/mock-item-2",
            readingTime: {
              text: "6 min read"
            }
          }
        }
      }
    ]
  };

  beforeAll(() => {
    jest.spyOn(console, "warn").mockImplementation(() => {});
  });

  it("should renders correctly", async () => {
    let featured = render(<FeaturedItem item={mockData} />);
    expect(featured).toMatchSnapshot();
  });
});
