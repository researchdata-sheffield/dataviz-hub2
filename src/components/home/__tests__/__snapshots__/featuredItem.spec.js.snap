// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home | Featured Items should renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex flex-wrap pb-6 bg-gray-900"
      >
        <div
          class="react-reveal group w-full sm:w-1/2 min-h-80 lg:min-h-95 xl:min-h-100 pb-28 overflow-hidden relative leading-none bg-gray-900 text-gray-300 flex-grow"
          tabindex="0"
        >
          <a
            gatsby="true"
            href="/blog/25/01/2021/mock-item-1"
          >
            <div
              class="flex top-0 left-0 absolute m-3 lg:m-5 text-white lg:bg-transparent lg:text-gray-300 group-hover:bg-gray-900 items-center rounded-lg p-1"
              style="z-index: 3;"
            >
              <svg
                class="text-yellow-300 text-2xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <p
                class="inline-block lg:opacity-0 group-hover:opacity-100 text-base font-semibold px-2"
                style="font-family: TUoS Blake;"
              >
                Featured
              </p>
            </div>
            <div
              class="sc-eCImPb hYxToh lg:greyScale-100 group-hover:greyScale-0 group-focus:greyScale-0 min-h-3/5 max-h-3/5"
              style="min-height: 420px;"
            />
            <div
              class="sc-gsDKAQ sc-dkPtRN bUCpIR fjUJCM pt-6 px-8 text-gray-600"
            >
              <h1
                class="mt-3 font-bold text-2xl 2xl:text-3xl leading-tight text-gray-200 group-hover:text-brand-blue overflow-y-hidden"
              >
                Mock item 1
              </h1>
              <p
                class="my-2 text-sm 2xl:text-lg leading-tight group-hover:text-gray-100"
              >
                Featured item mock.
              </p>
              <p
                class="my-2 text-xs 2xl:text-base pt-2 group-hover:text-gray-100"
              >
                <span>
                  Mock author 1
                </span>
              </p>
              <p
                class="my-2 text-xs 2xl:text-base text-gray-500 group-hover:text-gray-300"
              >
                25 January 2021
                 · 
                11 min read
              </p>
            </div>
          </a>
          <div
            class="px-8 lg:px-8 text-xs text-gray-500"
            style="font-weight: 400;"
          >
            <span>
              <a
                class="font-semibold hover:underline text-gray-500"
                gatsby="true"
                href="/blog/category/articles"
              >
                Articles
              </a>
               · 
            </span>
            <span>
              <a
                class="text-gray-500 hover:underline"
                gatsby="true"
                href="/blog/tag/dataviz"
              >
                Dataviz
              </a>
               · 
            </span>
            <span>
              <a
                class="text-gray-500 hover:underline"
                gatsby="true"
                href="/blog/tag/c"
              >
                C++
              </a>
               · 
            </span>
            <span>
              <a
                class="text-gray-500 hover:underline"
                gatsby="true"
                href="/blog/tag/interactive"
              >
                Interactive
              </a>
              
            </span>
          </div>
        </div>
        <div
          class="react-reveal group w-full sm:w-1/2 min-h-80 lg:min-h-95 xl:min-h-100 pb-28 overflow-hidden relative leading-none bg-gray-900 text-gray-300 flex-grow"
          tabindex="0"
        >
          <a
            gatsby="true"
            href="/blog/09/09/2020/mock-item-2"
          >
            <div
              class="flex top-0 left-0 absolute m-3 lg:m-5 text-white lg:bg-transparent lg:text-gray-300 group-hover:bg-gray-900 items-center rounded-lg p-1"
              style="z-index: 3;"
            >
              <svg
                class="text-yellow-300 text-2xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
              <p
                class="inline-block lg:opacity-0 group-hover:opacity-100 text-base font-semibold px-2"
                style="font-family: TUoS Blake;"
              >
                Featured
              </p>
            </div>
            <div
              class="sc-eCImPb hYxToh lg:greyScale-100 group-hover:greyScale-0 group-focus:greyScale-0 min-h-3/5 max-h-3/5"
              style="min-height: 420px;"
            />
            <div
              class="sc-gsDKAQ sc-dkPtRN bUCpIR fjUJCM pt-6 px-8 text-gray-600"
            >
              <h1
                class="mt-3 font-bold text-2xl 2xl:text-3xl leading-tight text-gray-200 group-hover:text-brand-blue overflow-y-hidden"
              >
                Mock featured item 2
              </h1>
              <p
                class="my-2 text-sm 2xl:text-lg leading-tight group-hover:text-gray-100"
              >
                Data visualizations have effects.
              </p>
              <p
                class="my-2 text-xs 2xl:text-base pt-2 group-hover:text-gray-100"
              >
                <span>
                  Mock author 2
                </span>
              </p>
              <p
                class="my-2 text-xs 2xl:text-base text-gray-500 group-hover:text-gray-300"
              >
                09 September 2020
                 · 
                6 min read
              </p>
            </div>
          </a>
          <div
            class="px-8 lg:px-8 text-xs text-gray-500"
            style="font-weight: 400;"
          >
            <span>
              <a
                class="font-semibold hover:underline text-gray-500"
                gatsby="true"
                href="/blog/category/articles"
              >
                Articles
              </a>
               · 
            </span>
            <span>
              <a
                class="text-gray-500 hover:underline"
                gatsby="true"
                href="/blog/tag/social-factors"
              >
                Social Factors
              </a>
               · 
            </span>
            <span>
              <a
                class="text-gray-500 hover:underline"
                gatsby="true"
                href="/blog/tag/effects"
              >
                Effects
              </a>
              
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex flex-wrap pb-6 bg-gray-900"
    >
      <div
        class="react-reveal group w-full sm:w-1/2 min-h-80 lg:min-h-95 xl:min-h-100 pb-28 overflow-hidden relative leading-none bg-gray-900 text-gray-300 flex-grow"
        tabindex="0"
      >
        <a
          gatsby="true"
          href="/blog/25/01/2021/mock-item-1"
        >
          <div
            class="flex top-0 left-0 absolute m-3 lg:m-5 text-white lg:bg-transparent lg:text-gray-300 group-hover:bg-gray-900 items-center rounded-lg p-1"
            style="z-index: 3;"
          >
            <svg
              class="text-yellow-300 text-2xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              />
            </svg>
            <p
              class="inline-block lg:opacity-0 group-hover:opacity-100 text-base font-semibold px-2"
              style="font-family: TUoS Blake;"
            >
              Featured
            </p>
          </div>
          <div
            class="sc-eCImPb hYxToh lg:greyScale-100 group-hover:greyScale-0 group-focus:greyScale-0 min-h-3/5 max-h-3/5"
            style="min-height: 420px;"
          />
          <div
            class="sc-gsDKAQ sc-dkPtRN bUCpIR fjUJCM pt-6 px-8 text-gray-600"
          >
            <h1
              class="mt-3 font-bold text-2xl 2xl:text-3xl leading-tight text-gray-200 group-hover:text-brand-blue overflow-y-hidden"
            >
              Mock item 1
            </h1>
            <p
              class="my-2 text-sm 2xl:text-lg leading-tight group-hover:text-gray-100"
            >
              Featured item mock.
            </p>
            <p
              class="my-2 text-xs 2xl:text-base pt-2 group-hover:text-gray-100"
            >
              <span>
                Mock author 1
              </span>
            </p>
            <p
              class="my-2 text-xs 2xl:text-base text-gray-500 group-hover:text-gray-300"
            >
              25 January 2021
               · 
              11 min read
            </p>
          </div>
        </a>
        <div
          class="px-8 lg:px-8 text-xs text-gray-500"
          style="font-weight: 400;"
        >
          <span>
            <a
              class="font-semibold hover:underline text-gray-500"
              gatsby="true"
              href="/blog/category/articles"
            >
              Articles
            </a>
             · 
          </span>
          <span>
            <a
              class="text-gray-500 hover:underline"
              gatsby="true"
              href="/blog/tag/dataviz"
            >
              Dataviz
            </a>
             · 
          </span>
          <span>
            <a
              class="text-gray-500 hover:underline"
              gatsby="true"
              href="/blog/tag/c"
            >
              C++
            </a>
             · 
          </span>
          <span>
            <a
              class="text-gray-500 hover:underline"
              gatsby="true"
              href="/blog/tag/interactive"
            >
              Interactive
            </a>
            
          </span>
        </div>
      </div>
      <div
        class="react-reveal group w-full sm:w-1/2 min-h-80 lg:min-h-95 xl:min-h-100 pb-28 overflow-hidden relative leading-none bg-gray-900 text-gray-300 flex-grow"
        tabindex="0"
      >
        <a
          gatsby="true"
          href="/blog/09/09/2020/mock-item-2"
        >
          <div
            class="flex top-0 left-0 absolute m-3 lg:m-5 text-white lg:bg-transparent lg:text-gray-300 group-hover:bg-gray-900 items-center rounded-lg p-1"
            style="z-index: 3;"
          >
            <svg
              class="text-yellow-300 text-2xl"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              />
            </svg>
            <p
              class="inline-block lg:opacity-0 group-hover:opacity-100 text-base font-semibold px-2"
              style="font-family: TUoS Blake;"
            >
              Featured
            </p>
          </div>
          <div
            class="sc-eCImPb hYxToh lg:greyScale-100 group-hover:greyScale-0 group-focus:greyScale-0 min-h-3/5 max-h-3/5"
            style="min-height: 420px;"
          />
          <div
            class="sc-gsDKAQ sc-dkPtRN bUCpIR fjUJCM pt-6 px-8 text-gray-600"
          >
            <h1
              class="mt-3 font-bold text-2xl 2xl:text-3xl leading-tight text-gray-200 group-hover:text-brand-blue overflow-y-hidden"
            >
              Mock featured item 2
            </h1>
            <p
              class="my-2 text-sm 2xl:text-lg leading-tight group-hover:text-gray-100"
            >
              Data visualizations have effects.
            </p>
            <p
              class="my-2 text-xs 2xl:text-base pt-2 group-hover:text-gray-100"
            >
              <span>
                Mock author 2
              </span>
            </p>
            <p
              class="my-2 text-xs 2xl:text-base text-gray-500 group-hover:text-gray-300"
            >
              09 September 2020
               · 
              6 min read
            </p>
          </div>
        </a>
        <div
          class="px-8 lg:px-8 text-xs text-gray-500"
          style="font-weight: 400;"
        >
          <span>
            <a
              class="font-semibold hover:underline text-gray-500"
              gatsby="true"
              href="/blog/category/articles"
            >
              Articles
            </a>
             · 
          </span>
          <span>
            <a
              class="text-gray-500 hover:underline"
              gatsby="true"
              href="/blog/tag/social-factors"
            >
              Social Factors
            </a>
             · 
          </span>
          <span>
            <a
              class="text-gray-500 hover:underline"
              gatsby="true"
              href="/blog/tag/effects"
            >
              Effects
            </a>
            
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
